framework/assets/yii.gridView.js

```diff
@@ -101,6 +101,9 @@
             $grid.find('form.gridview-filter-form').remove();
             var $form = $('<form action="' + url + '" method="get" class="gridview-filter-form" style="display:none" data-pjax></form>').appendTo($grid);
             $.each(data, function (name, value) {
+                if (value == '') {
+            		return;
```

Comment:
yeah. it is callback function, so it must end.

p.s.
not related to this topic -  still i have problem with removing model name from parameter for whole grid (i can remove it by setting custom filter per column one by one.. but that's not good, maybe you know how to do it in one place
i mean
[i]?BooksSearch[title]=zzzzzzzzz[/i]
to
[i]?title=zzzzzzzzz[/i]